wire.md

```diff
@@ -0,0 +1,1008 @@
+controls which fields will immediately follow this header.
+
+Clients encountering a `msg_type` they do not know how to parse MUST ignore and
+discard it.
+
+The `circuit_id` field is not currently in use, and MUST be set to all zeros.
+
+The request ID, `req_id`, is a 32-bit number, generated randomly by the
+requester. It is used to uniquely identify the request during its lifetime
+across the peers who may handle it. For networks with up to 43 million active
+requests within a single cabal, the probability of collision remains below 1%.
+(i.e. `(1 - 1 / (2**32)) ** 43_000_000 > 0.99`)
+
+When generating a new `req_id`, a client MUST discard and generate a new one if
+the one generated collides with a known active `req_id`.
+
+When a client forwards a request to further peers, the `req_id` MUST NOT be
+changed, so that routing loops can be more easily detected by peers in the
+network.
+
+The protocol MAY be extended by implementers by creating additional
+`msg_type`s. Implementers MUST only use `msg_type > 255`. The
+first 256 are reserved for core protocol use.
+
+#### 6.3.2 Requests
+
+##### 6.3.2.1 Header
+
+Every request MUST have its header bytes be proceeded by the following request
+header fields:
+
+field      | type       | desc
+-----------|------------|-----------------------------------
+`ttl`      | `varint`   | number of network hops remaining
+
+The value of `ttl` MUST be between 0 and 16. 0 means "do not forward this
+request to any other peers".
+
+More fields follow below for the different request types, whose bytes MUST
+immediately follow the above message and request header bytes.
+
+##### 6.3.2.2 Post Request
+
+Request a set of posts, given their hashes.
+
+field        | type                | desc
+-------------|---------------------|-------------------------------------
+`hash_count` | `varint`            | number of hashes to request
+`hashes`     | `u8[32*hash_count]` | hashes, concatenated together
+
+`msg_type` MUST be set to `2`.
+
+Results are provided by one or more Post Response messages.
+
+The responder SHOULD immediately return what data is locally available, rather
+than holding on to the request in anticipation of perhaps seeing the requested
+hashes in the future.
+
+##### 6.3.2.3 Cancel Request
+
+Conclude a given `req_id` and stop receiving responses for that request.
+
+This request can be used to terminated previously sent, long-lived requests.
+
+field        | type                | desc
+-------------|---------------------|-------------------------------------
+`cancel_id`  | `u8[4]`             | the `req_id` of the request to be cancelled
+
+Receiving this request indicates that any further responses sent back with a
+`req_id` matching the given `cancel_id` will be ignored and discarded.
+
+`msg_type` MUST be set to `3`.
+
+`cancel_id` MUST be set to the `req_id` of the request to be cancelled.
+
+There MUST NOT be any responses to this request.
```

Comment:
How would I respond to it in the first place? How would the receiver of that response detect that I'm responding? I hope they are not required to keep track of all cancellations they ever sent.

Reply:
Changed this to read, "There is no expected response to this request."